packages/server/lib/request.js

```diff
@@ -118,6 +118,11 @@ const maybeRetryOnNetworkFailure = function (err, options = {}) {
     // https://github.com/cypress-io/cypress/pull/6705
     debug('detected TLS version error, setting min version to TLSv1')
     opts.minVersion = 'TLSv1'
+
+    if (retryIntervals.length === 0) {
+      // normally, this request would not be retried, but we need to retry in order to support TLSv1
+      return onNext(0, 1)
```

Comment:
what does `onNext(0, 1)` do? Is there a way we could make the code more descriptive?

Reply:
https://github.com/cypress-io/cypress/blob/9e4c13a6c181689ee832606fd74e3c1b8ae1ce69/packages/server/lib/request.js#L284-L288

So this runs a single `retry` instantly.

> Is there a way we could make the code more descriptive?

So many ways. :smile: I started refactoring this file and it became a much bigger task, so I decided to keep this PR to the bare minimum changes. For now I pushed a commit to clean up `onNext`/`onElse` specifically: 5df0b86